### ISSUE TITLE
Contrib activity that sets S3 ACL

### ISSUE THREAD

**Issue by realstraw** [krux/hyperion#223]
`aws cli` does not provide very flexible ways to set s3 acl because of the following reasons:

1. `aws s3` does not allow mixing canned acl with `--grants`
1. `--grants` overwrites the existing acl instead of adding additional permissions.
1. `aws s3api put-object-acl` provides lower level access, but it does not have a `--recursive` option.

Add a contrib activity that allows more flexible managing of s3 acl with the following option

- `--acl canned-acl [canned-acl ...]` where `canned-acl` only accepts values of `private`, `public-read`, `public-read-write`, `authenticated-read`, `bucket-owner-read`, `bucket-owner-full-control` and `log-delivery-write`.
- `--grants Permission=Grantee_Type=Grantee_ID [Permission=Grantee_Type=Grantee_ID ...]` where grants format should be the same as the aws cli option http://docs.aws.amazon.com/cli/latest/reference/s3/cp.html
- `--overwrite` whether it overwrites the acl or adding new acl.

Basically, it works similar to `aws s3 cp` but allows specifying both `acl` and `grants`, where grants adds permission instead of replacing them.
